models/users.js

```diff
@@ -165,14 +165,16 @@ const fetchPaginatedUsers = async (query) => {
     const allUsers = [];
 
     snapshot.forEach((doc) => {
-      allUsers.push({
-        id: doc.id,
-        ...doc.data(),
-        phone: undefined,
-        email: undefined,
-        tokens: undefined,
-        chaincode: undefined,
-      });
+      if (!doc.data().roles.archived) {
+        allUsers.push({
+          id: doc.id,
+          ...doc.data(),
+          phone: undefined,
+          email: undefined,
+          tokens: undefined,
+          chaincode: undefined,
+        });
+      }
```

Comment:
since `doc.data()` is used at multiple places we can store it in a variable and use it at both places.

Reply:
done